product_available_by_warehouse/views/sale_view.xml

```diff
@@ -7,7 +7,7 @@
         <field name="model">sale.order</field>
         <field name="inherit_id" ref="sale_stock.view_order_form_inherit" />
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="before">
+            <xpath expr="//field[@name='order_line']/form//field[@name='product_id']" position="after">
```

Comment:
why are you changing the view? at least this is not mandatory in the requirement described in the task/issue.